include/libvfio-user.h

```diff
@@ -501,7 +501,13 @@ typedef struct {
     /*
      * Migration state transition callback.
      *
-     * The callback should return -1 on error, setting errno.
+     * The callback should return 0 on success and -errno on error.
```

Comment:
we moved to expecting -1 + errno, a while back

Reply:
I know, the problem is that setting `errno` in a Python callback when unit testing isn't visible in libvfio-user:
> The use_errno parameter, when set to true, enables a ctypes mechanism that allows accessing the system errno error number in a safe way. ctypes maintains a thread-local copy of the systems errno variable; if you call foreign functions created with use_errno=True then the errno value before the function call is swapped with the ctypes private copy, the same happens immediately after the function call.

Do you have any ideas how to solve this?

Reply:
I tested this, and you need to make sure your callback type *also* has "use_errno=True" e.g.

```
 vfu_region_access_cb_t = c.CFUNCTYPE(c.c_int, c.c_void_p, c.POINTER(c.c_char),
-                                     c.c_ulong, c.c_long, c.c_bool)
+                                     c.c_ulong, c.c_long, c.c_bool,
+                                     use_errno=True)
```

then you can use `c.set_errno()` inside the callback, and it should work OK

Reply:
Well that was easy, thanks!